File: z-clients/python/diom/apis/admin_auth_policy.py
# This file is @generated

from ..internal.api_common import ApiBase, parse_response
from ..models import (
    AdminAccessPolicyConfigureIn,
    AdminAccessPolicyConfigureOut,
    AdminAccessPolicyDeleteIn,
    AdminAccessPolicyDeleteOut,
    AdminAccessPolicyGetIn,
    AdminAccessPolicyListIn,
    AdminAccessPolicyOut,
    ListResponseAdminAccessPolicyOut,
)


class AdminAuthPolicyAsync(ApiBase):
    async def configure(
        self,
        admin_access_policy_configure_in: AdminAccessPolicyConfigureIn,
    ) -> AdminAccessPolicyConfigureOut:
        """Create or update an access policy"""
        body = admin_access_policy_configure_in.model_dump(exclude_none=True)

        response = await self._request_asyncio(
            method="post",
            path="/api/v1.admin.auth-policy.configure",
            body=body,
        )
        return parse_response(response, AdminAccessPolicyConfigureOut)

    async def delete(
        self,
        admin_access_policy_delete_in: AdminAccessPolicyDeleteIn,
    ) -> AdminAccessPolicyDeleteOut:
        """Delete an access policy"""
        body = admin_access_policy_delete_in.model_dump(exclude_none=True)

        response = await self._request_asyncio(
            method="post",
            path="/api/v1.admin.auth-policy.delete",
            body=body,
        )
        return parse_response(response, AdminAccessPolicyDeleteOut)

    async def get(
        self,
        admin_access_policy_get_in: AdminAccessPolicyGetIn,
    ) -> AdminAccessPolicyOut:
        """Get an access policy by ID"""
        body = admin_access_policy_get_in.model_dump(exclude_none=True)

        response = await self._request_asyncio(
            method="post",
            path="/api/v1.admin.auth-policy.get",
            body=body,
        )
        return parse_response(response, AdminAccessPolicyOut)

    async def list(
        self,
        admin_access_policy_list_in: AdminAccessPolicyListIn = AdminAccessPolicyListIn(),
    ) -> ListResponseAdminAccessPolicyOut:
        """List all access policies"""
        body = admin_access_policy_list_in.model_dump(exclude_none=True)

        response = await self._request_asyncio(
            method="post",
            path="/api/v1.admin.auth-policy.list",
            body=body,
        )
        return parse_response(response, ListResponseAdminAccessPolicyOut)


class AdminAuthPolicy(ApiBase):
    def configure(
        self,
        admin_access_policy_configure_in: AdminAccessPolicyConfigureIn,
    ) -> AdminAccessPolicyConfigureOut:
        """Create or update an access policy"""
        body = admin_access_policy_configure_in.model_dump(exclude_none=True)

        response = self._request_sync(
            method="post",
            path="/api/v1.admin.auth-policy.configure",
            body=body,
        )
        return parse_response(response, AdminAccessPolicyConfigureOut)

    def delete(
        self,
        admin_access_policy_delete_in: AdminAccessPolicyDeleteIn,
    ) -> AdminAccessPolicyDeleteOut:
        """Delete an access policy"""
        body = admin_access_policy_delete_in.model_dump(exclude_none=True)

        response = self._request_sync(
            method="post",
            path="/api/v1.admin.auth-policy.delete",
            body=body,
        )
        return parse_response(response, AdminAccessPolicyDeleteOut)

    def get(
        self,
        admin_access_policy_get_in: AdminAccessPolicyGetIn,
    ) -> AdminAccessPolicyOut:
        """Get an access policy by ID"""
        body = admin_access_policy_get_in.model_dump(exclude_none=True)

        response = self._request_sync(
            method="post",
            path="/api/v1.admin.auth-policy.get",
            body=body,
        )
        return parse_response(response, AdminAccessPolicyOut)

    def list(
        self,
        admin_access_policy_list_in: AdminAccessPolicyListIn = AdminAccessPolicyListIn(),
    ) -> ListResponseAdminAccessPolicyOut:
        """List all access policies"""
        body = admin_access_policy_list_in.model_dump(exclude_none=True)

        response = self._request_sync(
            method="post",
            path="/api/v1.admin.auth-policy.list",
            body=body,
        )
        return parse_response(response, ListResponseAdminAccessPolicyOut)

File: z-clients/rust/src/api/admin_auth_policy.rs
// this file is @generated
use crate::{Configuration, error::Result, models::*};

pub struct AdminAuthPolicy<'a> {
    cfg: &'a Configuration,
}

impl<'a> AdminAuthPolicy<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    /// Create or update an access policy
    pub async fn configure(
        &self,
        admin_access_policy_configure_in: AdminAccessPolicyConfigureIn,
    ) -> Result<AdminAccessPolicyConfigureOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.admin.auth-policy.configure")
            .with_body(admin_access_policy_configure_in)
            .execute(self.cfg)
            .await
    }

    /// Delete an access policy
    pub async fn delete(
        &self,
        admin_access_policy_delete_in: AdminAccessPolicyDeleteIn,
    ) -> Result<AdminAccessPolicyDeleteOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.admin.auth-policy.delete")
            .with_body(admin_access_policy_delete_in)
            .execute(self.cfg)
            .await
    }

    /// Get an access policy by ID
    pub async fn get(
        &self,
        admin_access_policy_get_in: AdminAccessPolicyGetIn,
    ) -> Result<AdminAccessPolicyOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.admin.auth-policy.get")
            .with_body(admin_access_policy_get_in)
            .execute(self.cfg)
            .await
    }

    /// List all access policies
    pub async fn list(
        &self,
        admin_access_policy_list_in: AdminAccessPolicyListIn,
    ) -> Result<ListResponseAdminAccessPolicyOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.admin.auth-policy.list")
            .with_body(admin_access_policy_list_in)
            .execute(self.cfg)
            .await
    }
}
